pkg/virt-api/webhooks/mutating-webhook/mutators/vm-mutator.go

```diff
@@ -97,6 +111,30 @@ func (mutator *VMsMutator) Mutate(ar *admissionv1.AdmissionReview) *admissionv1.
 	}
 }
 
+func (mutator *VMsMutator) applyFlavorToVm(vm *v1.VirtualMachine, oldVMRaw []byte) error {
+	if vm.UID == "" {
```

Comment:
Instead of using `UID` you can use `ar.Request.Operation` to check if the current operation is creation.